File: PatternTable/WordList.java
import java.util.ArrayList;

public class WordList {
  ArrayList<String> list = new ArrayList<>();

  public void insertWord(String str) {
    list.add(str);
  }

  public String nthWord(int n) {
    return list.get(n);
  }

  public int size() {
    return list.size();
  }

  ArrayList<String> getList() {
    return list;
  };

  public String toString() {
    String ans = "";

    for (String word : list) {
      ans += word + "\n";
    }

    return ans;
  }
}

File: WordList/WordList.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class WordList
{
  vector<string> list;

public:
  void insertWord(string str)
  {
    list.push_back(str);
  };

  string nthWord(int n)
  {
    if ((n - 1) > list.size())
    {
      throw out_of_range("out of range");
    }

    return list.at(n - 1);
  };

  int size()
  {
    return list.size();
  };

  vector<string> getList() {
    return list;
  }
};